.github/workflows/e2e.yml

```diff
@@ -199,14 +199,15 @@ jobs:
         uses: actions/cache@v3
         with:
           path: .next/cache
-          key: ${{ runner.os }}-next-cache-${{ github.sha }}
+          key: ${{ runner.os }}-${{ secrets.CACHE_VERSION }}-nextjs-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
           restore-keys: |
-            ${{ runner.os }}-next-cache-${{ github.sha }}
-            ${{ runner.os }}-next-cache-
+            {{ runner.os }}-${{ secrets.CACHE_VERSION }}-nextjs-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
+            ${{ runner.os }}-${{ secrets.CACHE_VERSION }}-nextjs-${{ hashFiles('**/package-lock.json') }}-
+            ${{ runner.os }}-${{ secrets.CACHE_VERSION }}-nextjs-
 
       - name: Build (frontend)
         if: steps.next-build.outputs.cache-hit != 'true'
-        run: npm run build
+        run: PRESERVE_NEXT_CACHE=true npm run build
```

Comment:
Good one, I actually missed that we were building here again. TBH I don't really understand: since this job depends on `build`, why do we need to rebuild here?

Reply:
Since `e2e` is a separate workflow, it does not share the workspace with `build`, the workspace starts clean and we can only get the pre-built artifact from cache.

It builds here again if there is no cache hit because its possible that the cache was evicted or this workflow was retried, etc.

Reply:
This cache use and optimization seems to indicate to me that maybe e2e should not be a separate workflow. Its now only triggered after the build step so I feel like we could again reorganize this so everything is on the same workflow for the purpose of sharing the previous step results.

For now, the reasoning here is that I saw some e2e jobs taking 5 to 6 minutes building the frontend because of the `.next/cache` key mismatch. We can just fix that and leave the build cache if that makes more sense to you.

https://github.com/opencollective/opencollective-frontend/actions/runs/5452957439/jobs/9921288047#step:26:39
![image](https://github.com/opencollective/opencollective-frontend/assets/5448927/b4106c91-ad37-4785-8e41-0b7734f8c19c)

Reply:
Yep that's already a huge improvement!